exdb/views.py

```diff
@@ -107,34 +109,52 @@ def get_success_url(self):
 
     def get_context_data(self, **kwargs):
         context = super(ExperienceApprovalView, self).get_context_data()
-        context['experience'] = self.get_experience()
+        if kwargs.get('invalid_comment'):
+            context['comment_form'] = kwargs.get('invalid_comment')
+        else:
+            context['comment_form'] = self.second_form_class()
         context['comments'] = ExperienceComment.objects.filter(experience=context['experience'])
         return context
 
-    def form_valid(self, form):
-        form.instance.author = self.request.user
-        form.instance.experience = self.get_experience()
-        form.instance.experience.status = 'ad' if self.request.POST.get('approve') else 'de'
-        if form.instance.experience.status == 'ad':
-            form.instance.experience.next_approver = None
-            ExperienceApproval.objects.create(
-                experience=form.instance.experience,
-                approver=self.request.user
-            )
-        form.instance.experience.save()
-        return super(ExperienceApprovalView, self).form_valid(form)
-
-    def form_invalid(self, form):
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        experience_form = self.get_form()
+        comment_form = self.second_form_class(request.POST)
+        if experience_form.is_valid() and comment_form.is_valid():
```

Comment:
Change the above condition to:
```python
if experience_form.is_valid() and (self.request.POST.get('approve') or comment_form.is_valid()):
```
Because the other logic path diminishes the meaning of `form_invalid()` which is confusing. You'll have to update the callees (`form_valid()`, `form_invalid()`, and maybe `get_context_data()`) to make everything make sense again.